takanon_extractions/transform_to_markdown.py

```diff
@@ -0,0 +1,702 @@
+                    if not response or not response.choices:
+                        raise ValueError("Empty response from OpenAI API")
+                    
+                    # Get the content from the first choice
+                    content: str = response.choices[0].message.content
+                    if not content:
+                        raise ValueError("Empty content in response")
+                    
+                    # Remove markdown code block markers if present
+                    content = content.strip()
+                    if content.startswith("```markdown"):
+                        content = content[11:].strip()
+                    elif content.startswith("```"):
+                        content = content[3:].strip()
+                    if content.endswith("```"):
+                        content = content[:-3].strip()
+                    
+                    logger.info("Markdown conversion completed successfully")
+                    return content
+                    
+                except Exception as e:
+                    error_msg: str = str(e)
+                    if attempt < max_retries - 1:  # Don't retry on the last attempt
+                        delay: int = base_delay * (2 ** attempt)  # Exponential backoff
+                        logger.warning(f"Attempt {attempt + 1} failed: {error_msg}. Retrying in {delay} seconds...")
+                        time.sleep(delay)
+                    else:
+                        raise  # Re-raise the last exception
+                        
+        except Exception as e:
+            error_msg: str = str(e)
+            logger.error(f"Error converting text to markdown: {error_msg}")
+            
+            if "authentication" in error_msg.lower():
+                logger.error("Authentication error. Please check your OPENAI_API_KEY environment variable.")
+            elif "rate limit" in error_msg.lower():
+                logger.error("Rate limit exceeded. Please try again later or check your API quota.")
+            elif "model not found" in error_msg.lower():
+                logger.error(f"Model {self.model} not found. Please check if the model name is correct.")
+            else:
+                logger.error("Unknown error occurred. Please check your internet connection and API key.")
+            
+            raise
+
+    def split_text(self, text: str, max_chunk_size: int = 4000) -> List[str]:
+        """Split text into chunks of approximately equal size, trying to break at natural boundaries.
+        
+        Args:
+            text: The text to split
+            max_chunk_size: Maximum size of each chunk in characters
+            
+        Returns:
+            List of text chunks
+        """
+        chunks: List[str] = []
+        current_chunk: List[str] = []
+        current_size: int = 0
+        
+        # Split text into paragraphs
+        paragraphs: List[str] = text.split('\n\n')
```

Comment:
If we're working with HTMLs, this would be useless, no? There are no such empty lines in an html (and if so, not necessarily meaning a paragraph break).

Reply:
So how do we get the text files for the laws then?